plutus-tx/src/PlutusTx/Blueprint/TH.hs

```diff
@@ -40,10 +44,90 @@ makeHasSchemaInstance dataTypeName indices = do
   let nonOverlapInstance = TH.InstanceD Nothing
 
   -- Lookup indices for all constructors of a data type.
-  indexedCons <- for (TH.datatypeCons dataTypeInfo) $ \ctorInfo ->
-    case lookup (TH.constructorName ctorInfo) indices of
-      Just index -> pure (ctorInfo, index)
-      Nothing    -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+  indexedCons :: [(TH.ConstructorInfo, SchemaInfo, Natural)] <- do
+    typeSchemaAnns <- lookupSchemaAnns dataTypeName
+    for (TH.datatypeCons dataTypeInfo) $ \ctorInfo -> do
+      let ctorName = TH.constructorName ctorInfo
+      case lookup ctorName indices of
+        Nothing -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+        Just index -> do
+          {- Its possible to annotate a type as well as its constructors with schema annotations.
+             Consider the following example:
+
+            {-# ANN MkLeft (SchemaDescription "Constructor: MkLeft") #-}
+            {-# ANN MkRight (SchemaDescription "Constructor: MkRight") #-}
+            data Or = MkLeft | MkRight
+
+            With these annotations the following schema will be generated:
+
+            { ...
+              "Or": {
+                "oneOf": [
+                  {
+                    "description": "Constructor: MkLeft",
+                    "dataType": "constructor",
+                    "fields": [],
+                    "index": 0
+                  },
+                  {
+                    "description": "Constructor: MkRight",
+                    "dataType": "constructor",
+                    "fields": [],
+                    "index": 1
+                  }
+                ]
+              },
+            ...
+            }
+
+            Such schema lists all constructors of a type with their respective annotations,
+            however, there is no place to put annotations for the type itself.
+
+            For this reason, we merge annotations for the type and its constructor:
```

Comment:
What reason? the previous sentence doesn't justify _merging_ them. In particular, I don't see from this what's wrong with my suggestion: for _all_ of the type and the constructors, use the "title" field for the name, and the "description" field for the documentation. No merging.

Reply:
> What reason? the previous sentence doesn't justify merging them. 

I didn't expand on the reason enough:
By "there is no place to put annotations for the type itself."  I meant "there is no place to put annotations for the type itself, and one would need to repeat a type-specific annotation for every constructor, which is counter-ergonomic and should be avoided".

Reply:
After discussing this more with @michaelpj we came to an agreement  to go forward without such functionality to merge type-level/ctor-level annotations. This was an attempt to work around the blueprint-schema limitation that "oneOf" can't have adjacent attributes: "title", "description", etc.